hypothesis-ruby/src/lib.rs

```diff
@@ -151,13 +152,24 @@ methods!(
         name: RString,
         database_path: RString,
         seed: Integer,
-        max_example: Integer
+        max_example: Integer,
+        skip_phases: Array
     ) -> AnyObject {
+        let rust_skip_phases = safe_access(skip_phases).into_iter().filter_map(|ruby_phase| {
+            let phase_sym = safe_access(ruby_phase.try_convert_to::<Symbol>());
+            if let "shrink" = phase_sym.to_str() {
```

Comment:
The current approach is clear and concise, but what do you think about moving that logic to `TryFrom` when there will be more phases & more consumers that will do conversion (python?)? I.e. if `Phase` would implement `TryFrom` for `&str`? Then it might look like this:

```rust
safe_access(skip_phases).into_iter().filter_map(|ruby_phase| safe_access(ruby_phase.try_convert_to::<Symbol>()).to_str().try_into().ok()).collect()
```

Reply:
However, in the Python version, there is an enum & [a guard](https://github.com/HypothesisWorks/hypothesis/blob/master/hypothesis-python/src/hypothesis/_settings.py#L519) against passing invalid phases. Would it make sense to have the same for Ruby?